pontoon/translations/views.py

```diff
@@ -22,6 +22,31 @@
 from pontoon.translations import forms
 
 
+def _send_badge_notification(user, badge, level):
+    desc = """
+        You have gained a new badge level!
+        <br>
+        {badge}: Level {level}
+        <br>
+        You can view this badge on your <a href={profile_href}> profile page </a>.
+        """.format(
+        badge=badge,
+        level=level,
+        profile_href=reverse(
+            "pontoon.contributors.contributor.username",
+            kwargs={
+                "username": user,
+            },
+        ),
+    )
+    notify.send(
+        sender=user,
+        recipient=user,
+        verb="ignore",  # Triggers render of description only
+        description=desc,
+    )
```

Comment:
Good idea to factor this code out! 

I'd suggest to use a template to format the notification, like here:
https://github.com/mozilla/pontoon/blob/0d15df6dc01cda21a97c2dbb90cceb7b8a73d2c9/pontoon/projects/management/commands/send_suggestion_notifications.py#L118-L125

Also, `translations` module is not the best place to keep this code, because it's also used for promotions. Let's at least add a TODO item to the PR description to file an issue before this PR is merged to move the code to pontoon.messaging, where I'd like to keep all emails and notifications in one place.